src/Http/Livewire/LivewireDatatable.php

```diff
@@ -516,14 +530,29 @@ public function setSessionStoredFilters()
 
     public function initialiseSort()
     {
-        $this->sort = $this->defaultSort()
-        ? $this->defaultSort()['key']
-        : collect($this->freshColumns)->reject(function ($column) {
-            return in_array($column['type'], Column::UNSORTABLE_TYPES) || $column['hidden'];
-        })->keys()->first();
+        $this->sort = ($columns = $this->defaultSort())->isNotEmpty()
+            ? array_values($columns->transform(function ($column) {
+                return $column['key'] . '|' . $column['direction'];
+            })->toArray())
```

Comment:
Let's only use collection methods instead of mixing in native php array manipulation functions.

```suggestion
            ? $columns->transform(function ($column) {
                return $column['key'] . '|' . $column['direction'];
            })->toArray()->values()
```